cloud_testing/kotlin_poc/src/main/kotlin/ftl/gc/GcAndroidTestMatrix.kt

```diff
@@ -0,0 +1,79 @@
+package ftl.gc
+
+import com.google.api.services.testing.Testing
+import com.google.api.services.testing.model.*
+import ftl.config.YamlConfig
+import ftl.util.Utils.fatalError
+import ftl.util.Utils.join
+import ftl.util.Utils.uniqueObjectName
+import java.util.concurrent.TimeUnit
+
+object GcAndroidTestMatrix {
+
+    fun build(
+            appApkGcsPath: String,
+            testApkGcsPath: String,
+            runGcsPath: String,
+            androidMatrix: AndroidMatrix,
+            testShardsIndex: Int = -1,
+            config: YamlConfig): Testing.Projects.TestMatrices.Create {
+        val testShardsTotal = config.testShardChunks.size
+
+        if (testShardsIndex >= testShardsTotal) {
+            throw RuntimeException("Invalid test shard index $testShardsIndex not < $testShardsTotal")
+        }
+
+        // https://github.com/bootstraponline/studio-google-cloud-testing/blob/203ed2890c27a8078cd1b8f7ae12cf77527f426b/firebase-testing/src/com/google/gct/testing/launcher/CloudTestsLauncher.java#L120
+        val testMatrix = TestMatrix()
+        testMatrix.clientInfo = ClientInfo().setName("Flank")
+
+        // Use default instrumentationTestRunner
+        //   String instrumentationTestRunner = "";
+        //   .setTestRunnerClass(instrumentationTestRunner)
+
+        val androidInstrumentation = AndroidInstrumentationTest()
+                .setAppApk(FileReference().setGcsPath(appApkGcsPath))
+                .setTestApk(FileReference().setGcsPath(testApkGcsPath))
+
+        if (config.useOrchestrator) {
+            androidInstrumentation.orchestratorOption = "USE_ORCHESTRATOR"
+        }
+
+        androidInstrumentation.testTargets = config.testShardChunks[testShardsIndex]
+
+        val testTimeoutSeconds = TimeUnit.MINUTES.toSeconds(config.testTimeoutMinutes)
+
+        // --auto-google-login
+        // https://cloud.google.com/sdk/gcloud/reference/firebase/test/android/run
+        // https://github.com/bootstraponline/gcloud_cli/blob/e4b5e01610abad2e31d8a6edb20b17b2f84c5395/google-cloud-sdk/lib/googlecloudsdk/api_lib/firebase/test/android/matrix_creator.py#L174
+        var account: Account? = null
+
+        if (config.autoGoogleLogin) {
+            account = Account().setGoogleAuto(GoogleAuto())
+        }
+
+        val testSetup = TestSetup()
+                .setDirectoriesToPull(listOf("/sdcard/screenshots"))
```

Comment:
kt version still doesn't have `directories-to-pull`?

Reply:
At the moment, the kt version has almost no configuration options. I opened a ticket for directories to pull: 
https://github.com/TestArmada/flank/issues/185

The goal is to eventually support all the configuration options as the gcloud cli.